templates/checkout.php

```diff
@@ -25,11 +25,18 @@
 	$timestamp = time();
 
 	$context_params = array(
-		'plugin_id'         => $fs->get_id(),
-		'plugin_public_key' => $fs->get_public_key(),
-		'plugin_version'    => $fs->get_plugin_version(),
+		'plugin_id'      => $fs->get_id(),
+		'plugin_version' => $fs->get_plugin_version(),
+        'public_key'     => $fs->get_public_key(),
+        'mode'           => 'dashboard',
+        'plan_id'        => $_GET['plan_id'],
+        'trial'          => ( isset( $_GET['trial'] ) ? $_GET['trial'] : false ),
```

Comment:
@fajardoleo same thing here, you have to validate that the `trial` is boolean. If I'm not mistaken, you actually need to check if it's a string and one of the following `true` or `false`, because there's no real way to pass booleans via querystring request. So you need to use `in_array( $_GET['trial'], array( 'true', 'false' ) )`.